src/Evaluator.hs

```diff
@@ -23,6 +80,14 @@ import Syntax.SrcLoc
 import Type
 import Value
 
+-- -----------------------------------------------------------------------------
+-- Interpreter Data Types
+
+
+data EvalResult
+  = ExampleResult SrcLoc VEnv Core (Scheme Ty) Value
+  | IOResult (IO ())
```

Comment:
`EvalResult` is not even used in this file :sweat_smile: 
Perhaps it should be defined next to `evalMod` instead?

Let's do that in a separate MR, as this MR does not introduce `EvalResult`, it merely moves it to the top of the file.